File: 05_hash_tables/scala/02_check_voter.scala
object checkVoter {
  def main(args: Array[String]): Unit = {
    import scala.collection.mutable
    val voted = mutable.Map[String, Boolean]()
    def checkVoter(name: String): Unit = {
      if (voted.exists(_ == (name, true))) {
        println("kick them out!")
      } else {
        voted += (name -> true)
        println("let them vote!")
      }
    }
    check_voter("tom")
    check_voter("mike")
    check_voter("mike")
  }
}

File: 05_hash_tables/kotlin/02_check_voter.kt
private val voted = hashMapOf<String, Boolean>()

private fun checkVoter(name: String) {
    println(
        if (voted.containsKey(name)) {
            "kick them out!"
        } else {
            voted[name] = true
            "let them vote!"
        }
    )
}

fun main(args: Array<String>) {
    checkVoter("tom")   // let them vote!
    checkVoter("mike")  // let them vote!
    checkVoter("mike")  // kick them out!
}